runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowMetrics.java

```diff
@@ -209,7 +209,14 @@ private BoundedTrieResult getBoundedTrieValue(MetricUpdate metricUpdate) {
       if (metricUpdate.getTrie() == null) {
         return BoundedTrieResult.empty();
       }
-      BoundedTrie bTrie = (BoundedTrie) metricUpdate.getTrie();
+      BoundedTrie bTrie;
+      try {
+        bTrie = (BoundedTrie) metricUpdate.getTrie();
+      } catch (ClassCastException cce) {
+        LOG.warn(
+            "Failed to retrieve BoundedTrie metrics from result. Returning empty result.", cce);
+        return BoundedTrieResult.empty();
```

Comment:
I'm not super comfortable with returning an incorrect result here due to a class cast exception...